### PR TITLE
BUG: pickling ArrowExtensionArray with _pa_data rename

### PR DESCRIPTION
**Pull request by mroeschke**
xref https://github.com/pandas-dev/pandas/pull/50987#discussion_r1134365537

### CODE REVIEW

**Review by phofl — APPROVED** [review]
thx @mroeschke

File: pandas/core/arrays/arrow/array.py [hunk 1]
```diff
@@ -433,11 +433,15 @@ def __abs__(self) -> Self:
...
     def __getstate__(self):
         state = self.__dict__.copy()
-        state["_data"] = self._pa_array.combine_chunks()
+        state["_pa_array"] = self._pa_array.combine_chunks()
         return state
 
     def __setstate__(self, state) -> None:
-        state["_pa_array"] = pa.chunked_array(state["_data"])
+        if "_data" in state:
+            data = state["_data"]
```
Comment (jbrockmendel) [inline pandas/core/arrays/arrow/array.py:441]: i think pop it here to avoid including it in `__dict__.update` below?

Comment (mroeschke) [reply pandas/core/arrays/arrow/array.py:441]: Ah good call. Thanks!

**Comment by phofl** [comment]
thx @mroeschke

**Pull request merged**